airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/ctid/CtidGlobalStateManager.java

```diff
@@ -50,25 +51,34 @@ public CtidGlobalStateManager(final CtidStreams ctidStreams,
     this.stateManager = stateManager;
     this.savedOffsetAfterReplicationSlotLSN = savedOffsetAfterReplicationSlotLSN;
     this.defaultCdcState = defaultCdcState;
-    initStream(ctidStreams, catalog);
     this.fileNodeHandler = fileNodeHandler;
+    initStream(ctidStreams, catalog);
   }
 
   private void initStream(final CtidStreams ctidStreams,
                           final ConfiguredAirbyteCatalog catalog) {
     this.streamsThatHaveCompletedSnapshot = new HashSet<>();
     this.resumableFullRefreshStreams = new HashSet<>();
+    this.nonResumableFullRefreshStreams = new HashSet<>();
     catalog.getStreams().forEach(configuredAirbyteStream -> {
+      var pair =
+          new AirbyteStreamNameNamespacePair(configuredAirbyteStream.getStream().getName(), configuredAirbyteStream.getStream().getNamespace());
       if (!ctidStreams.streamsForCtidSync().contains(configuredAirbyteStream) && configuredAirbyteStream.getSyncMode() == SyncMode.INCREMENTAL) {
-        streamsThatHaveCompletedSnapshot.add(
-            new AirbyteStreamNameNamespacePair(configuredAirbyteStream.getStream().getName(), configuredAirbyteStream.getStream().getNamespace()));
+        streamsThatHaveCompletedSnapshot.add(pair);
       }
-      if (ctidStreams.streamsForCtidSync().contains(configuredAirbyteStream)
+
+      if (fileNodeHandler.hasFileNode(
+          new AirbyteStreamNameNamespacePair(configuredAirbyteStream.getStream().getName(), configuredAirbyteStream.getStream().getNamespace()))
           && configuredAirbyteStream.getSyncMode() == SyncMode.FULL_REFRESH) {
-        this.resumableFullRefreshStreams.add(
-            new AirbyteStreamNameNamespacePair(configuredAirbyteStream.getStream().getName(), configuredAirbyteStream.getStream().getNamespace()));
+        this.resumableFullRefreshStreams.add(pair);
+      } else if (configuredAirbyteStream.getSyncMode() == SyncMode.FULL_REFRESH) {
+        this.nonResumableFullRefreshStreams.add(pair);
```

Comment:
This can be refactored to be the following:
```
if (configuredAirbyteStream.getSyncMode() == SyncMode.FULL_REFRESH) {
  if (fileNodeHandler.hasFileNode(
          new AirbyteStreamNameNamespacePair(configuredAirbyteStream.getStream().getName(), configuredAirbyteStream.getStream().getNamespace()))
    this.resumableFullRefreshStreams.add(pair);
  else
    this.nonResumableFullRefreshStreams.add(pair);
}
```